library/gcse/src/gcse_page.e

```diff
@@ -0,0 +1,113 @@
+note
+	description: "Represent metadata describing the query for the current set of results."
+	date: "$Date: 2015-10-09 08:11:07 -0300 (vi., 09 oct. 2015) $"
+	revision: "$Revision: 97973 $"
+
+class
+	GCSE_PAGE
+
+feature -- Access
+
+	search_terms: detachable STRING_8
+			-- search term
+
+	title: detachable STRING_8
+			-- Search title.
+
+	total_results: INTEGER
+			-- Search total results.
+
+	count: INTEGER
+			-- Rows per page.
+
+	start_index: INTEGER
+			-- Page index.
+
+feature -- Element change
+
+	set_search_terms (a_search_terms: like search_terms)
+			-- Assign `search_terms' with `a_search_terms'.
+		do
+			search_terms := a_search_terms
+		ensure
+			search_terms_assigned: search_terms = a_search_terms
+		end
+
+feature -- Change element
+
+	set_title (a_title: like title)
+			-- Set title with `a_title'
+		do
+			title := a_title
+		ensure
+			title_set: title = a_title
+		end
+
+	set_total_results (a_total_results: like total_results)
+			-- Set total_results with `a_total_results'.
+		do
+			total_results := a_total_results
+		ensure
+			total_results_set: total_results = a_total_results
+		end
+
+	set_count (a_count: like count)
+			-- Set count with `a_count'.
+		do
+			count := a_count
+		ensure
+			count_set: count = a_count
+		end
+
+	set_start_index (a_start_index: like start_index)
+			-- Set start_index with `a_start_index'.
+		do
+			start_index := a_start_index
+		ensure
+			start_index_set: start_index = a_start_index
+		end
+
+feature -- Output
+
+		to_string: STRING_8
```

Comment:
Is this used? or mainly for debugging purpose?
if just for debugging .. maybe you can use  debug_output: ...